test/parallel/test-vm-module-basic.js

```diff
@@ -85,7 +85,11 @@ const util = require('util');
 
   assert.strictEqual(
     m[util.inspect.custom].call(Object.create(null)),
```

Comment:
honestly this should validate the receiver and throw if it isn't correct.

Reply:
I added the same instance check that's used in the other properties. It won't prevent manipulation that would allow to change the context to something else though.

Currently `URL`, `TextDecoder` and `Module` have fallbacks for null constructors that have no corresponding test (e.g. https://coverage.nodejs.org/coverage-b46bbf2e1990999f/lib/internal/url.js.html#L364). It seems like it should be prevented by the instance check but the check is not that restrictive that it prevents all cases. Do you know if this is intentional?

Reply:
I am not certain about the semverness of adding the validation. It should probably be semver-major, even though it's unlikely to apply.

Reply:
If it's considered semver-major, I'd just remove the validation again.